src/hotspot/cpu/x86/x86.ad

```diff
@@ -3335,7 +3450,18 @@ instruct sqrtD_reg(regD dst) %{
   ins_pipe(pipe_slow);
 %}
 
+
 // ---------------------------------------- VectorReinterpret ------------------------------------
+instruct reinterpret_mask(kReg dst) %{
+  predicate(UseAVX > 2 && vector_length(n) == vector_length(n->in(1))); // dst == src
+  match(Set dst (VectorReinterpret dst));
+  ins_cost(125);
+  format %{ "vector_reinterpret $dst\t!" %}
+  ins_encode %{
+    // empty
+  %}
+  ins_pipe( pipe_slow );
+%}
```

Comment:
This is overriding the instructs reinterpret_expand* and reitnerpret_shrink. So doesnt look correct.

Reply:
Both have different predication condition.

Reply:
ok